### PR TITLE
As 92 synonyms

### PR DESCRIPTION
**Pull request by YuriHechter**

### CODE REVIEW

File: cmd/index.go [hunk 1]
```diff
@@ -207,3 +220,139 @@ func restartSearchLogsFn(cmd *cobra.Command, args []string) {
...
+			log.Error(errors.Wrapf(err, "Error at scanning synonym config file: %s.", filePath))
+			return
+		}
+
+		//log.Printf("Keywords: %v", keywords)
+		synonymsBody := fmt.Sprintf(bodyMask, "synonym_graph", "synonym_graph", strings.Join(keywords, ","))
+
+		//log.Printf("Update synonyms request body: %v", body)
+
+		// Close the index in order to update the synonyms
```
Comment (bbfsdev) [inline cmd/index.go]: What close does?

File: config.sample.toml [hunk 2]
```diff
@@ -25,6 +25,8 @@ sources-folder="/tmp/sources-folder"
 unzip-url=""
 prepare-docs-batch-size=20
 prepare-docs-parallelism=2
+synonyms-folder="$GOPATH/src/github.com/Bnei-Baruch/archive-backend/es/synonyms"
+#index-date = "2018-11-28t13:08:31-05:00" # optional
```
Comment (bbfsdev) [inline config.sample.toml]: Why did you add it?
Why not as flag?

File: es/mappings/make.py [hunk 3]
```diff
@@ -143,15 +143,15 @@ def IsCyrillic(lang, something):
...
-        "he": {
-          "tokenizer": "standard",
-          "filter": [
-            "he_IL"
-          ],
-          "char_filter": [
-            "quotes"
-          ]
-        },
+        #"he": {
```
Comment (bbfsdev) [inline es/mappings/make.py]: Why commented out?

File: es/mappings/make.py [hunk 4]
```diff
@@ -162,6 +162,13 @@ def IsCyrillic(lang, something):
         #     "custom_phonetic",
         #   ],
         # },
+        "synonym" : {
```
Comment (bbfsdev) [inline es/mappings/make.py]: Why we have something special for he_IL?

File: search/engine.go [hunk 5]
```diff
@@ -222,7 +222,7 @@ func (e *ESEngine) AddIntents(query *Query, preference string, size int, sortBy
 	potentialIntents := make([]Intent, 0)
 	for _, language := range query.LanguageOrder {
 		// Order here provides the priority in results, i.e., tags are more importnt then sources.
-		index := es.IndexAliasName("prod", consts.ES_RESULTS_INDEX, language)
+		index := es.IndexNameByDefinedDateOrAlias("prod", consts.ES_RESULTS_INDEX, language)
```
Comment (bbfsdev) [inline search/engine.go]: Why did you switch the way we load index?

File: search/query.go [hunk 6]
```diff
@@ -40,29 +40,29 @@ func createResultsQuery(resultTypes []string, q Query, docIds []string) elastic.
...
+					elastic.NewMatchQuery("content.language", q.Term).Analyzer("synonym"),
 				).MinimumNumberShouldMatch(1),
 			).Boost(0.0),
 		).Should(
 			elastic.NewDisMaxQuery().Query(
 				// Language analyzed
-				elastic.NewMatchPhraseQuery("title.language", q.Term).Slop(SLOP).Boost(TITLE_BOOST),
-				elastic.NewMatchPhraseQuery("description.language", q.Term).Slop(SLOP).Boost(DESCRIPTION_BOOST),
-				elastic.NewMatchPhraseQuery("content.language", q.Term).Slop(SLOP),
+				elastic.NewMatchPhraseQuery("title.language", q.Term).Analyzer("synonym").Slop(SLOP).Boost(TITLE_BOOST),
```
Comment (bbfsdev) [inline search/query.go]: Does this defines the language?
How that works for different languages?
I thought the analyzer was chosen per field?
Please explain?

is this query time analyzer? which in different from index time?

File: search/query.go [hunk 7]
```diff
@@ -178,7 +178,7 @@ func NewResultsSearchRequests(options SearchRequestOptions) []*elastic.SearchReq
 	requests := make([]*elastic.SearchRequest, 0)
 	indices := make([]string, len(options.query.LanguageOrder))
 	for i := range options.query.LanguageOrder {
-		indices[i] = es.IndexAliasName("prod", consts.ES_RESULTS_INDEX, options.query.LanguageOrder[i])
+		indices[i] = es.IndexNameByDefinedDateOrAlias("prod", consts.ES_RESULTS_INDEX, options.query.LanguageOrder[i])
```
Comment (bbfsdev) [inline search/query.go]: Why not alias?

File: cmd/index.go [hunk 8]
```diff
@@ -207,3 +220,113 @@ func restartSearchLogsFn(cmd *cobra.Command, args []string) {
...
+			return
+		}
+		if !closeRes.Acknowledged {
+			log.Errorf("CloseIndex not Acknowledged: %s", indexName)
+			return
+		}
+
+		defer openIndex(indexName)
+		encodedIndexName := url.QueryEscape(indexName)
+		_, err = common.ESC.PerformRequest(context.TODO(), elastic.PerformRequestOptions{
```
Comment (bbfsdev) [inline cmd/index.go]: Why not use the API:
https://godoc.org/github.com/olivere/elastic#IndicesPutSettingsService
Please update.

File: es/mappings/make.py [hunk 9]
```diff
@@ -109,6 +117,120 @@
...
+    }
+  },
+  SPANISH: {
+    "spanish_synonym": {
+            "tokenizer":  "standard",
+            "filter": [
+              "lowercase",
+              "spanish_stop",
+              "spanish_keywords",
+              "spanish_stemmer"
```
Comment (bbfsdev) [inline es/mappings/make.py]: Synonym graph?!

File: search/data/en Recall Set - Recall set en.csv [hunk 10]
```diff
@@ -4,7 +4,7 @@ en,toronto,7.5,קבוצה,"Page showing Toronto group details: address, contact
 en,arvut,328.3,נושא בקבלה,https://kabbalahmedia.info/sources/itcVAcFn,https://kabbalahmedia.info/lessons?source=bs_qMeV5M3Y_itcVAcFn,Excerpts on topic: Arvut,,,
```
Comment (bbfsdev) [inline search/data/en Recall Set - Recall set en.csv]: Please explain all the recall set changes?
Did you change them also in spreadsheet?

File: search/query.go [hunk 11]
```diff
@@ -147,17 +147,23 @@ func NewResultsSearchRequest(options SearchRequestOptions) *elastic.SearchReques
 		Size(options.size).
 		Explain(options.query.Deb)
 
+	//
```
Comment (bbfsdev) [inline search/query.go]: Remove

File: es/mappings/make.py [hunk 12]
```diff
@@ -109,6 +117,120 @@
...
+
+LanguageFiltersImp ={
+  ENGLISH: {
+            "english_stop": {
+              "type":       "stop",
+              "stopwords":  "_english_" 
+            },
+            "english_keywords": {
+              "type":       "keyword_marker",
+              "keywords":   ["example"]
```
Comment (bbfsdev) [inline es/mappings/make.py]: This example and in other languages looks like a mistake. Not sure why it is here.

File: cmd/index.go (hunk 8 above)
Comment (YuriHechter) [reply cmd/index.go]: done

File: search/query.go (hunk 11 above)
Comment (YuriHechter) [reply search/query.go]: done

File: es/mappings/make.py (hunk 12 above)
Comment (YuriHechter) [reply es/mappings/make.py]: done

File: es/mappings/make.py (hunk 9 above)
Comment (YuriHechter) [reply es/mappings/make.py]: added

**Comment by bbfsdev** [comment]
Manually merged.
